readme.md

```diff
@@ -3,37 +3,58 @@
 
+* [What is MapStruct?](#what-is-mapstruct)
+* [Requirements](#requirements)
+* [Using MapStruct](#using-mapstruct)
+ * [Maven](#maven)
+ * [Gradle](#gradle)
+* [Documentation and getting help](#documentation-and-getting-help)
+* [Licensing](#licensing)
+* [Building from Source](#building-from-source)
+* [Links](#links)
+
 ## What is MapStruct?
 
-MapStruct is a Java [annotation processor](http://docs.oracle.com/javase/6/docs/technotes/guides/apt/index.html) for the generation of type-safe bean mapping classes.
+MapStruct is a Java [annotation processor](http://docs.oracle.com/javase/6/docs/technotes/guides/apt/index.html) for the generation of type-safe and performant mappers for Java bean classes.
+
+To create a mapping between two types, declare a mapper class like this:
+
+```java
+@Mapper
+public interface CarMapper {
 
-All you have to do is to define a mapper interface which declares any required mapping methods. During compilation, MapStruct will generate an implementation of this interface. This implementation uses plain Java method invocations for mapping between source and target objects, i.e. no reflection or similar.
+    CarMapper INSTANCE = Mappers.getMapper( CarMapper.class );
 
-Compared to writing mapping code from hand, MapStruct saves time by generating code which is tedious and error-prone to write. Following a convention over configuration approach, MapStruct uses sensible defaults but steps out of your way when it comes to configuring or implementing special behavior.
+    @Mapping(source = "numberOfSeats", target = "seatCount")
+    CarDto carToCarDto(Car car);
+}
+```
 
-Compared to dynamic mapping frameworks, MapStruct offers the following advantages:
+At compile time MapStruct will generate an implementation of this interface. The generated implementation uses plain Java method invocations for mapping between source and target objects, i.e. there is no reflection involved. By default, properties are mapped if they have the same name in source and target, but this and many other aspects can be controlled using `@Mapping` and a handful of other annotations.
+
+MapStruct saves you from writing mapping code by hand which is a tedious and error-prone task. The generator comes with sensible defaults and many built-in type conversions, but it steps out of your way when it comes to configuring or implementing special behavior.
+
+Compared to mapping frameworks working at runtime MapStruct offers the following advantages:
 
 * Fast execution by using plain method invocations instead of reflection
 * Compile-time type safety: Only objects and attributes mapping to each other can be mapped, no accidental mapping of an order entity into a customer DTO etc.
 * Self-contained code, no runtime dependencies
-* Clear error-reports at build time, if entities or attributes can't be mapped
+* Clear error-reports at build time if mappings are incorrect and incomplete
```

Comment:
sub-bullits. 
1. incomplete (not all target properties are mapped).
2. incorrect (cannot find a proper mapping method / conversion).

I've come to really appreciate 1. as one of our strongest assets.